aikau/src/main/resources/alfresco/core/Events.js

```diff
@@ -43,163 +43,172 @@ define(["dojo/_base/declare",
-          * @param {string} timeout variable name containing timeout.
-          * @param {string} topic event to trigger
-          * @param {int} triggerInterval Optional override for this.triggerInterval
-          * @param {bool} throttle switches to throttle mode.
-          */
-         _limit: function alfresco_core_Events__debounce(timeout, topic, triggerInterval, throttle) {
-            // If we've already got a timeout, we need to rate limit
-            if (this[timeout]) {
-               if (throttle) {
-                  // In throttle mode, ignore subsequent events while timeout exists
-                  return;
-               }
-               // In debounce mode, clear timeout and let a new one get set.
-               clearTimeout(this[timeout]);
+   return declare([AlfCore, EventsTopicMixin], {
+
+      /**
+       * How frequently do we want rate limited events to trigger?
+       *
+       * @instance
+       * @type {int}
+       * @default 300
+       */
+      triggerInterval: 300,
+
+      /**
+       * A specific element to detect the scroll position of. If this is left as the default of null
+       * then the overall window will be used.
+       *
+       * @instance
+       * @type {element}
+       * @default null
+       */
+      scrollElement: null,
+
+      /**
+       * Used to store a handle to cancel the scroll timeout.
+       *
+       * @instance
+       * @type {int}
+       * @default: null
+       */
+      scrollTimeout: null,
+
+      /**
+       * Used to store a handle to cancel the resize timeout.
+       *
+       * @instance
+       * @type {int}
+       * @default: null
+       */
+      resizeTimeout: null,
+
+      /**
+       * The scroll event triggers once when the page had loaded. Generally we don't need this.
+       * Should we ignore it?
+       *
+       * @instance
+       * @type Boolean
+       * @default true
+       */
+      ignoreInitialScroll: true,
+
+      /**
+       *  Has the initial scroll event as specified in ignoreInitialScroll happened yet?
+       */
+      hasInitialScrollHappened: false,
+
+      /**
+       * Registers the event listeners.
+       *
+       * @instance
+       */
+      registerEventListeners: function alfresco_core_Events__registerEventListeners() {
+         this.alfLog("log", "Registering listeners...");
+
+         var targetElement = this.scrollElement || window;
+         on(targetElement, "scroll", lang.hitch(this, this.onScroll));
+
+         // TODO: Sweep through modules and replace individual resize listeners with this?
+         on(targetElement, "resize", lang.hitch(this, this.onResize));
+      },
+
+      /**
+       * The limit function. Prevents multiple events triggering in very quick succession.
+       *
+       * In Throttle mode:
+       * Triggers once per triggerInterval for as long as the event keeps being fired.
+       * TODO: In throttle mode should event trigger before or after delay? (currently after)
```

Comment:
I didn't write the code originally, nor did I write the comment, this issue doesn't require addressing those - I've simply moved some code around.